matplotlib2tikz/legend.py

```diff
@@ -181,6 +187,17 @@ def draw_legend(data, obj):
             alignment = None
             break
 
+    # Set color of lines in legend
+    data['legend colors'] = []
+    for handle in obj.legendHandles:
+        try:
+            data, legend_color, _ = mycol.mpl_color2xcolor(data,
+                                                           handle.get_color())
+            data['legend colors'].append('\\addlegendimage{no markers, %s}\n'
+                                         % legend_color)
+        except:
```

Comment:
> Why do we need an exception type?

Check out https://stackoverflow.com/q/14797375/353337 for some details.

> fails in case of polar plots.

Ah, funny. And otherwise it's okay? What fails exactly?